src/jmath/moveWithCollision.ts

```diff
@@ -42,6 +43,7 @@ export function isForceMoveProjectile(x: ForceMove): x is ForceMoveProjectile {
 
 interface ForceMoveProjectileArgs {
   pushedObject: HasSpace;
+  source: any;
```

Comment:
let's type these to something specific, not `any`

Reply:
It can be an union type if needed
https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#union-types